src/truncate.jl

```diff
@@ -49,33 +49,94 @@ insupport(d::Truncated{D,Union{Discrete,Continuous}}, x::Real) where {D<:Univari
 
 quantile(d::Truncated, p::Real) = quantile(d.untruncated, d.lcdf + p * d.tp)
 
-for f in [:pdf, :logpdf, :cdf, :logcdf, :ccdf, :logccdf]
-    @eval ($f)(d::Truncated, x::Int) = ($f)(d, float(x))
+function _pdf(d::Truncated, x::T) where {T<:Real}
```

Comment:
Do you really need to define these underscored functions? Is that to avoid ambiguities?

Reply:
(see main thread)